### ISSUE TITLE
bindings to concrete types not propagating correctly from template to collection

### ISSUE THREAD

**Issue by sjmiles** [Polymer/polymer#1839]
Given:

```html
     <template is="dom-repeat" items="{{somethings}}">
        <paper-input value="{{item}}"></paper-input>
      </template>
```

changes to `paper-input.value`appear in `template.getModelForElement(paper-input).item` but not in `template.items[n]`.

http://jsbin.com/qocuka/2/edit

**Comment by pvmart** [comment]
Might be related.
Overwriting an object in array with the new one destroys binding.
``` js
         this.push('test', {value: 1});
         this.set('test.0.value',  2);
         this.set('test.0',{value: 3});
         this.set('test.0.value',  4);
```
The result is 3 but in the model is 4.
http://plnkr.co/edit/AH5SKoedUQgeyfBgsfWL?p=preview

**Comment by kevinpschaaf** [comment]
@pvmart Yep, they are the same root cause.  A fix is in the works that should solve both this and https://github.com/Polymer/polymer/issues/1854.

**Comment by kevinpschaaf** [comment]
This should be fixed by PR #1855.

Confirmed it fixes repro case here:
http://jsbin.com/ziseru/2/edit

**Comment by kevinpschaaf** [comment]
Root cause here w.r.t. notifying based on individual `dom-repeat` instances' `item` changng was fixed in #1854.

Note related issue opened here: https://github.com/Polymer/polymer/issues/1913, regarding problems when primitive values are duplicated in the array.
